### PR TITLE
feat: Fee utils

### PR DESCRIPTION
**Pull request by ermvrs**
[](https://github.com/keep-starknet-strange/satoru/issues/265)

# Pull Request type

Please add the labels corresponding to the type of changes your PR introduces:

- Bugfix
- Feature
- Code style update (formatting, renaming)
- Refactoring (no functional changes, no API changes)
- Build-related changes
- Documentation content changes
- Testing
- Other (please describe):

## What is the current behavior?

Resolves: #NA

## What is the new behavior?

-
-
-

### CODE REVIEW

File: src/event/event_emitter.cairo [hunk 1]
```diff
@@ -239,5 +319,58 @@ mod EventEmitter {
...
+            fee_type: felt252
+        ) {
+            self
+                .emit(
+                    ClaimableUiFeeAmountUpdated {
+                        ui_fee_receiver, market, token, delta, next_value, next_pool_value, fee_type
+                    }
+                );
+        }
+
```
Comment (sparqet) [inline src/event/event_emitter.cairo:352]: add comment with the event emitted

File: src/event/event_emitter.cairo [hunk 2]
```diff
@@ -239,5 +319,58 @@ mod EventEmitter {
...
+        fn emit_fees_claimed(
+            ref self: ContractState,
+            market: ContractAddress,
+            receiver: ContractAddress,
+            fee_amount: u128
+        ) {
+            self.emit(FeesClaimed { market, receiver, fee_amount });
+        }
+
```
Comment (sparqet) [inline src/event/event_emitter.cairo:361]: add comment with the event emitted

File: src/event/event_emitter.cairo [hunk 3]
```diff
@@ -239,5 +319,58 @@ mod EventEmitter {
...
+            ref self: ContractState,
+            market: ContractAddress,
+            token: ContractAddress,
+            delta: u128,
+            next_value: u128,
+            fee_type: felt252
+        ) {
+            self.emit(ClaimableFeeAmountUpdated { market, token, delta, next_value, fee_type });
+        }
+
```
Comment (sparqet) [inline src/event/event_emitter.cairo:334]: add comment with the event emitted

**Review by sparqet — COMMENTED** [review]
Looks good, you just need to add 3 commets and it will be merged !

File: src/event/event_emitter.cairo (hunk 3 above)
Comment (ermvrs) [reply src/event/event_emitter.cairo:334]: Done 👍

**Pull request merged**